### ISSUE TITLE
Implicit class with two arguments does not fail compilation

### ISSUE THREAD

**Issue by nicolasstucki** [lampepfl/dotty#3540]
```scala
object Test {
  implicit class Foo(i: Int, s: String)
}
```
Fails in `scalac` with
```
Foo.scala:2: error: implicit classes must accept exactly one primary constructor parameter
  implicit class Foo(i: Int, s: String)
                 ^
one error found
```

But not on `dotc`. This error should probably be emitted when desugaring the `implicit class`.

**Comment by vitorsvieira** [comment]
I'd like to give it a try. 

Would you recommend to work on the #2464 first?

-V

**Comment by allanrenucci** [comment]
I would start with this one. It might as well fix #2464

**Comment by vitorsvieira** [comment]
OK. Thanks, Allan.

**Comment by nicolasstucki** [comment]
Yes, you should start with this one. 
#2464 has some additional complex interaction with phantom types. If you fix this one you will partially #2464, at least for 99.99 of the use cases. But do not close that one.
